### DISCUSSION TITLE
Notification Service / Hooks

### DISCUSSION THREAD

**Discussion by nbs2904** [felixhoffmnn/aswe#17] [Ideas]
Zusätzlich zu der Möglichkeit den Agenten aktiv Dinge zu fragen und Antworten zu erhalten, sollte dieser auch von sich aus die Möglichkeit haben den Nutzer über besondere Ereignisse zu informieren:

- Störmeldungen im Verkehr
- Breaking News
- Wetter Warnungen

Hierbei müssten wir wahrscheinlich parallel einen Subservice laufen lassen, der entweder bestimmte Hooks abonniert oder konstant nach gewissen Ereignissen sucht.

Wie / Wann binden wir diesen am besten ein?
Welche APIs eignen sich dafür, wenn überhaupt?

**Comment by felixhoffmnn** [comment]
Mein aktueller Plan ist, das Skript dauerhaft zuhören zu lassen (wie bei Alexa). Daher könnten wir in einem regelmäßigen Abstand Requests senden. Jedoch würde das auf Dauer viel Bandbreite brauchen (ist also nicht optimal).

**Reply by nbs2904** [reply]
Zusätzlich rennen wir eventuell so in ein Request Limit rein. 
Optimal wären wirklich Hooks bei denen wir unseren Service registrieren könnten. Notfalls lassen wir den Notification Service dann begrenzt laufen.
